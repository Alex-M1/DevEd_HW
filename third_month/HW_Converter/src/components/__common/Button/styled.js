import styled from 'styled-components';

export const StButton = styled.button`
  padding: 5px 10px;
  background: transparent;
  border: 2px solid #4583ff;
  border-radius: 4px;
  cursor: pointer;
  &:hover{
    background: #4583ff;
    color: #fff
  }
`;
