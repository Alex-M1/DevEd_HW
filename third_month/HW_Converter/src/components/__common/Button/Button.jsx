import React from 'react';
import PropTypes from 'prop-types';
import { StButton } from './styled';

function Button({ onClick }) {
  return (
    <StButton onClick={onClick}>
      Convert
    </StButton>
  );
}

export default Button;

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
};
