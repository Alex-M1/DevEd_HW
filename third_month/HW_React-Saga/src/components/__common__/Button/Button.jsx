import React from 'react';
import PropTypes from 'prop-types';
import { StButton } from './styled';

function Button({ children, type, onClick }) {
  return (
    <StButton
      mode={type}
      onClick={onClick}
    >
      {children}
    </StButton>
  );
}

export default Button;

Button.propTypes = {
  type: PropTypes.string,
  onClick: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired,
};
