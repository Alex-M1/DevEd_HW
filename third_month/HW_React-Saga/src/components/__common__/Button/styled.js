import styled from 'styled-components';
import { btnStyles } from '../../../helpers';

export const StButton = styled.button`
  background:${({ mode }) => btnStyles[mode].bg};
  color: #fff;
  padding: 8px;
  border: none;
  border-radius: 5px;
  cursor: pointer;
  min-width: 80px;
  &:hover{
    background:${({ mode }) => btnStyles[mode].hoverBg};
  }
`;
