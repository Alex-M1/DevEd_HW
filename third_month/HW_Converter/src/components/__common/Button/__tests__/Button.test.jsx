import { mount, shallow } from 'enzyme';
import React from 'react';
import Button from '../Button';

describe('Form', () => {
  const props = { onClick: jest.fn() };
  it('Should match snapshot', () => {
    const component = shallow(<Button {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have StButton', () => {
    const component = mount(<Button {...props} />);
    expect(component.find('styled__StButton')).toHaveLength(1);
  });
  it('Should have call onClick', () => {
    const component = mount(<Button {...props} />);
    component.find('styled__StButton').simulate('click');
    expect(component.props().onClick).toHaveBeenCalledTimes(1);
  });
});
