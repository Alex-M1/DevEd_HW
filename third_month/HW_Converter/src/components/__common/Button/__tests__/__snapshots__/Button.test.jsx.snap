// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Should match snapshot 1`] = `"<button class=\\"styled__StButton-sc-z42suv-0 dyLAGI\\">Convert</button>"`;
